src/app/content/highlights/utils/getHighlightLocationFilters.ts

```diff
@@ -8,16 +9,26 @@ import {
 } from '../../utils/archiveTreeUtils';
 import { HighlightLocationFilters } from '../types';
 
-const getHighlightLocationFilters = (book: Book | ArchiveBook | undefined) => {
-  return !book
-    ? new Map() as HighlightLocationFilters
-    : new Map(
-      flattenArchiveTree(book.tree)
-        .filter((section) =>
-          (archiveTreeSectionIsPage(section) && archiveTreeSectionIsBook(section.parent))
-          || (archiveTreeSectionIsChapter(section) && !archiveTreeSectionIsUnit(section)))
-        .map((section) => [section.id, section])
-    );
-};
+type LocationFilterSection = LinkedArchiveTree | LinkedArchiveTreeSection;
 
-export default getHighlightLocationFilters;
+const sectionIsLocationFilter = (section: LocationFilterSection) =>
+  (archiveTreeSectionIsPage(section) && archiveTreeSectionIsBook(section.parent))
+  || (archiveTreeSectionIsChapter(section) && !archiveTreeSectionIsUnit(section));
+
+const getLocationFilterSectionsForBook = (book: Book | ArchiveBook | undefined) => book
+  ? flattenArchiveTree(book.tree).filter(sectionIsLocationFilter)
+  : [];
+
+const sectionsToLocationFilters = (sections: LocationFilterSection[]): HighlightLocationFilters =>
+  new Map(sections.map((section) => [section.id, section]));
+
+export const getHighlightLocationFilters = flow(
```

Comment:
it seems weird to make the SG logic (which is actually simpler) depend on all this complex logic for highlights. what if we make `getHighlightLocationFilters` accept the `filterBy` option all the time, and move the  `sectionIsLocationFilter` function into wherever the highlights logic is calling this.

Reply:
I moved the location-filters-related utils into one file because the one used as filter for highlights didn't feel like it belongs anywhere and it was used in few tests.